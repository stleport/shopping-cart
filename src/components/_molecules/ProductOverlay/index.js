import React from 'react';
import PropTypes from 'prop-types';
import { pluralize } from '../../../utils';

const ProductOverlay = ({ quantity }) => (
  (
    <div>
      <div className="st-Card__overlay">
        <h3>
          {quantity}
          &#160;portion
          {pluralize(quantity)}
          &#160;dans votre Frichti
        </h3>
      </div>
    </div>
  )
);

ProductOverlay.defaultProps = {
  quantity: 0
};

ProductOverlay.propTypes = {
  quantity: PropTypes.number
};

export default ProductOverlay;
