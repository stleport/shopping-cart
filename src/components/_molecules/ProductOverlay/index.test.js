import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ProductOverlay from '.';

function ProductOverlaySetup(newProps = {}) {
  const props = {
    quantity: 2
  };

  const ProductOverlayJSX = (
    <ProductOverlay {...props} />
  );

  const ProductOverlayComponent = shallow(ProductOverlayJSX);
  return {
    ProductOverlayComponent,
    ProductOverlayJSX,
    props
  };
}

describe('ProductOverlay', () => {
  it('renders correctly ProductOverlay component', () => {
    const { ProductOverlayJSX } = ProductOverlaySetup();
    const ProductOverlayComponent = renderer.create(ProductOverlayJSX).toJSON();
    expect(ProductOverlayComponent).toMatchSnapshot();
  });

  it('should pluralize "portion" when needed', () => {
    const { ProductOverlayComponent } = ProductOverlaySetup();
    expect(ProductOverlayComponent.find('h3').text()).toMatch(/portions/);
  });
});
