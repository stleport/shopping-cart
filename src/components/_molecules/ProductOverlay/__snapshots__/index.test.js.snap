// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductOverlay renders correctly ProductOverlay component 1`] = `
<div
  className="st-Card__overlay"
>
  <h3>
    2
     portion
    s
     dans votre Frichti
  </h3>
</div>
`;
